### PR TITLE
Populate repository add/edit fields with websocket data

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done
- Populated RepositoryAdd and RepositoryEdit forms with packagerepository and general data

## QA
- Go to `:8000/MAAS/settings/repositories` and click "Add PPA".
- Check that the url field is already prefixed with `ppa:`, there is no section for "Distributions" or "Components", the title is "Add PPA" and the save button says "Save PPA"
- Click on "Add repository"
- Check that there is a section for "Distributions" and "Components", the title is "Add repository" and the save button says "Save repository"
- Add some repos at `:5240/MAAS/#/settings/repositories`
- Go to `:8000/MAAS/settings/repositories`, click the edit icon and check that the repo data is correct
- Click the edit button for the first repo (default one)
- Check that there are checkboxes for "Disabled pockets" and "Disabled components"

### Add PPA
10 153 61 90_8000_MAAS_settings_repositories_add_ppa

### Add repository
10 153 61 90_8000_MAAS_settings_repositories_add_ppa (1)

### Edit PPA
10 153 61 90_8000_MAAS_settings_repositories (2)

### Edit repository
10 153 61 90_8000_MAAS_settings_repositories_add_ppa (2)

### Edit default repository
10 153 61 90_8000_MAAS_settings_repositories (3)

### Mobile view
10 153 61 90_8000_MAAS_settings_repositories (4)

Fixes canonical-web-and-design/maas-squad#1431
Fixes canonical-web-and-design/maas-squad#1433

### CODE REVIEW

**Comment by lilyvidenova** [comment]
@Caleb-Ellis on the mobile view, could we move the Enable repository and Enable sources checkboxes to just under the URL field? Other than that - looks good!

**Comment by Caleb-Ellis** [comment]
@lilyana Updated the position of the enable checkboxes. Also, forgot to make Name and URL required fields. ALSO, disabled Name and URL if repo is default - I think being able to edit those in old MAAS is a mistake.

**Comment by cassiocassio** [comment]
minor tings:

- (perhaps one for the backlog), I would like to systematically add links into docs across settings for this kind of screen. In this case https://maas.io/docs/package-repositories being one click away would be useful.


- How long is a key? It's useful if you can see the start and end of a key when you paste, to know you got it all,  which seems to be impossible in the old design (see docs) and obv. is unlikely in the desktop view

**Comment by lilyvidenova** [comment]
The key and all other similar fields grow to the size of the content in them, so you'd be able to see all of it at any time.

File: ui/src/app/settings/views/Repositories/RepositoryForm/RepositoryForm.js [hunk 1]
```diff
@@ -50,34 +70,101 @@ export const RepositoryForm = ({ title }) => {
...
+                    disabled_pockets: values.disabled_pockets,
+                    distributions: values.distributions
+                      .split(" ,")
+                      .filter(Boolean),
+                    enabled: values.enabled,
+                    key: values.key,
+                    name: values.name,
+                    url: values.url
+                  };
+                  console.log(payload);
```
Comment (squidsoup) [inline ui/src/app/settings/views/Repositories/RepositoryForm/RepositoryForm.js]: Stray `console.log` here

File: ui/src/app/settings/views/Repositories/RepositoryForm/RepositoryForm.js [hunk 2]
```diff
@@ -50,34 +70,101 @@ export const RepositoryForm = ({ title }) => {
...
+                  <Form onSubmit={formikProps.handleSubmit}>
+                    <RepositoryFormFields
+                      formikProps={formikProps}
+                      type={type}
+                    />
+                    <div className="repo-form__buttons">
+                      <Button
+                        appearance="base"
+                        className="u-no-margin--bottom"
+                        onClick={() => window.history.back()}
```
Comment (squidsoup) [inline ui/src/app/settings/views/Repositories/RepositoryForm/RepositoryForm.js]: 90% sure that you don't want to use `window.history` directly and instead should use the [react-router history api](https://reacttraining.com/react-router/web/api/history), but as there's currently no hooks support, perhaps we want to give [use-react-router](https://github.com/CharlesStover/use-react-router) a go?

File: ui/src/app/settings/views/Repositories/RepositoryForm/RepositoryForm.js (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/settings/views/Repositories/RepositoryForm/RepositoryForm.js]: @squidsoup Replaced with an alert :P. It's just there temporarily until #160 is implemented

File: ui/src/app/settings/views/Repositories/RepositoryForm/RepositoryForm.js (hunk 2 above)
Comment (Caleb-Ellis) [reply ui/src/app/settings/views/Repositories/RepositoryForm/RepositoryForm.js]: @squidsoup Yeah it felt wrong. Anyway we already had a custom `useRouter` hook in `base/hooks.js` that uses the same context as `use-react-router` does under the hood, so I've used that. There's already a comment to replace them when official `react-router` hooks are available.

**Pull request merged**